servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2603,6 +2603,55 @@ public Template getTemplate(ResourceContext ctx, String templateName) {
         return template;
     }
 
+    @Override
+    public DomainTemplateDetailsList getDomainTemplateDetailsList(ResourceContext ctx, String domainName) {
+        DomainTemplateDetailsList domainTemplateDetailsList = new DomainTemplateDetailsList();
+        List<TemplateMetaData> templateMetaDataList = new ArrayList<>();
+        Map<String, Integer> templateDomainMapping = new HashMap<>();
```

Comment:
these should be created after the validation of data otherwise we're just wasting resources if the request is invalid

Reply:
Done.